File: comms/bulk-messaging/senders/resolve-sender/resolve-sender.py
from twilio_comms import TwilioComms, CommunicationRecipient
from twilio_comms.senders import SendersResolveRequestSenderPoolId

client = TwilioComms(
    account_id="<username>",
    auth_token="<password>",
)

client.senders.resolve(
    request=SendersResolveRequestSenderPoolId(
        recipient_addresses=[
            CommunicationRecipient(
                address="+14153902337",
                channel="PHONE",
            ),
            CommunicationRecipient(
                address="+14153902337",
                channel="WHATSAPP",
            ),
            CommunicationRecipient(
                address="davidpletnjov@example.com",
                channel="EMAIL",
            )
        ],
    ),
)

File: comms/bulk-messaging/senders/resolve-sender/resolve-sender.php
<?php

namespace Example;

use Twilio\Comms\TwilioComms;
use Twilio\Comms\Senders\Types\SendersResolveRequestSenderPoolId;
use Twilio\Comms\Types\CommunicationRecipient;
use Twilio\Comms\Types\CommunicationChannel;

$client = new TwilioComms(
    accountId: '<username>',
    authToken: '<password>',
);
$client->senders->resolve(
    new SendersResolveRequestSenderPoolId([
        'recipientAddresses' => [
            new CommunicationRecipient([
                'address' => '+14153902337',
                'channel' => CommunicationChannel::Phone->value,
            ]),
            new CommunicationRecipient([
                'address' => '+14153902337',
                'channel' => CommunicationChannel::Whatsapp->value,
            ]),
            new CommunicationRecipient([
                'address' => 'davidpletnjov@example.com',
                'channel' => CommunicationChannel::Email->value,
            ]),
        ],
    ]),
);
